bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java

```diff
@@ -243,7 +243,11 @@ public String toString() {
      */
     @Override
     public void stateChanged(Item item, State oldState, State newState) {
+        State oldGroupState = this.state;
         setState(function.calculate(members));
+        if (!oldGroupState.equals(this.state)) {
+            sendGroupStateChangedEvent(item.getName(), this.state, oldGroupState);

```

Comment:
If we rely on the fast, that stateUpdated is always called before a stateChanged, why is "setState(function.calculate(members));" called in this function? It is already called in stateUpdated(), too.